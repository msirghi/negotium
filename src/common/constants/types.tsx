import React from 'react';
import { OverridableComponent } from '@mui/types';
import { SvgIconTypeMap } from '@mui/material';

export type MainMenuItem = {
  title: string;
  Icon: OverridableComponent<SvgIconTypeMap<{}, 'svg'>> & { muiName: string };
};
