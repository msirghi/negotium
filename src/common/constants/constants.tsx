import { MainMenuItem, ProjectColor, SiteTheme } from './types';
import StarBorderIcon from '@mui/icons-material/StarBorder';
import InboxIcon from '@mui/icons-material/Inbox';
import DateRangeIcon from '@mui/icons-material/DateRange';
import DoneIcon from '@mui/icons-material/Done';
import { SnackbarOrigin } from 'notistack';
import colors from '../styles/colors';
import { Descendant } from 'slate';

export const MAIN_MENU_ITEMS: MainMenuItem[] = [
  {
    Icon: InboxIcon,
    title: 'pageTitles.inbox',
    route: '/home/inbox',
  },
  {
    Icon: StarBorderIcon,
    title: 'pageTitles.today',
    route: '/home/today',
  },
  {
    Icon: DateRangeIcon,
    title: 'pageTitles.upcoming',
    route: '/home/upcoming',
  },
  // {
  //   Icon: DoneIcon,
  //   title: 'Completed',
  //   route: '/home/completed',
  // },
];

export const PROJECT_NAME = 'Negotium';

export const BASE_API_URL = 'http://localhost:4000';

export const BASE_API_URL_V1 = 'https://negotium-api.herokuapp.com/api/v1';

export const MAX_PROJECT_LIST_COUNT = 5;

export const SNACKBAR_POSITIONS: { [key: string]: SnackbarOrigin } = {
  BOTTOM_CENTER: { horizontal: 'center', vertical: 'bottom' },
};

export const PROJECT_COLORS: ProjectColor[] = [
  {
    color: 'rgb(219, 64, 53)',
    name: 'Red',
  },
  {
    color: 'rgb(255, 153, 51)',
    name: 'Orange',
  },

  {
    color: 'rgb(250, 208, 0)',
    name: 'Yellow',
  },
];

export const MENTION_ARRAY_KEYWORDS = ['Today', 'Tomorrow'];

export const SUPPORTED_LANGUAGES = [
  {
    code: 'en',
    title: 'English',
  },
  {
    code: 'ru',
    title: 'Russian',
  },
];

export const HOME_VIEW_LIST = [
  {
    title: 'Inbox',
    url: '/home/inbox',
  },
  {
    title: 'Today',
    url: '/home/today',
  },
];

export const TIME_FORMATS = [
  {
    title: '13:00',
    key: '24h',
  },
  {
    title: '1pm',
    key: 'am/pm',
  },
];

export const tempAccountInfo = {
  id: '1',
  email: 'mihail.sirghi@gmail.com',
  name: 'Mihail',
};

export const pagesWithoutWrapper = ['/login', '/registration'];

export const passwordStatusColors = {
  tooWeak: 'red',
  weak: 'orange',
  medium: 'purple',
  strong: 'green',
};

export const siteThemes: SiteTheme[] = [
  {
    color: colors.primaries.lightBlue_1,
    label: 'Default',
    internalKey: 'default',
  },
  {
    color: colors.black,
    label: 'Noir',
    internalKey: 'noir',
  },
];

export const initialRichTextValue: Descendant[] = [
  {
    type: 'paragraph',
    children: [{ text: '' }],
  },
];

export const ACTIVATION_CODE_VAR_NAME = 'ac';
