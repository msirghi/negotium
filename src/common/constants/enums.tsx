export enum PASSWORD_STRENGTH_STATUS {
  TOO_WEAK = 'Too weak',
  WEAK = 'Weak',
  MEDIUM = 'Medium',
  STRONG = 'Strong',
}

export enum ROW_DIRECTION {
  ROW = 'row',
  COLUMN = 'column',
}

export enum ACCOUNT_SETTINGS_CHANGE_MODE {
  EMAIL = 'EMAIL',
  PASSWORD = 'PASSWORD',
}

export enum HttpMethod {
  DELETE = 'DELETE',
  GET = 'GET',
  HEAD = 'HEAD',
  OPTIONS = 'OPTIONS',
  PATCH = 'PATCH',
  POST = 'POST',
  PUT = 'PUT',
}
