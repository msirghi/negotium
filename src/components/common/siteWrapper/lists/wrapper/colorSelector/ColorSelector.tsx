import Box from '@mui/material/Box';
import MenuItem from '@mui/material/MenuItem';
import FormControl from '@mui/material/FormControl';
import { useState } from 'react';
import { makeStyles } from '@mui/styles';
import { PROJECT_COLORS } from '../../../../../../common/constants/constants';
import { Row } from '../../../../utilities/row/Row';
import { TextField } from '@mui/material';

const useStyles = makeStyles({
  root: {
    marginTop: 24,
  },
  colorCircle: {
    borderRadius: '50%',
    width: 10,
    height: 10,
  },
  colorName: {
    marginLeft: 10,
  },
});

export const ColorSelector = () => {
  const [selectedColor, setSelectedColor] = useState(PROJECT_COLORS[0].color);
  const classes = useStyles();

  const handleChange = (event: { target: { value: string } }) => {
    setSelectedColor(event.target.value);
  };

  return (
    <Box className={classes.root}>
      <FormControl fullWidth>
        <TextField
          select
          size={'small'}
          value={selectedColor}
          label="Color"
          onChange={handleChange}
        >
          {PROJECT_COLORS.map(({ color, name }) => {
            return (
              <MenuItem key={color} value={color}>
                <Row alignVerticalCenter>
                  <div
                    className={classes.colorCircle}
                    style={{ backgroundColor: color }}
                  />
                  <span className={classes.colorName}>{name}</span>
                </Row>
              </MenuItem>
            );
          })}
        </TextField>
      </FormControl>
    </Box>
  );
};
