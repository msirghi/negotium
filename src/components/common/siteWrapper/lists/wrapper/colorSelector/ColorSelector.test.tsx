import renderer from 'react-test-renderer';
import { ColorSelector } from './ColorSelector';
import { mount } from 'enzyme';
import { TextField } from '@mui/material';
import { act } from '@testing-library/react';
import { PROJECT_COLORS } from '../../../../../../common/constants/constants';

describe('ColorSelector', () => {
  const defaultProps = {};

  it('should match the snapshot', () => {
    const tree = renderer.create(<ColorSelector {...defaultProps} />);
    expect(tree).toMatchSnapshot();
  });

  it('should set the data on menu item click', () => {
    const wrapper = mount(<ColorSelector {...defaultProps} />);
    const selector = wrapper.find(TextField).at(0) as any;
    act(() => {
      selector.props().onChange({ target: { value: PROJECT_COLORS[0].name } });
    });
    wrapper.update();
    expect(selector.props().value).toEqual(PROJECT_COLORS[0].color);
  });
});
