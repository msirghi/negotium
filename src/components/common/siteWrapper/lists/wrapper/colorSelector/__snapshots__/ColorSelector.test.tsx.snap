// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorSelector should match the snapshot 1`] = `
<div
  className="makeStyles-root-1 MuiBox-root css-0"
>
  <div
    className="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
  >
    <div
      className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
    >
      <label
        className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1kty9di-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink={true}
      >
        Color
      </label>
      <div
        className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall MuiSelect-root css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        onClick={[Function]}
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          className="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-jedpe8-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          role="button"
          tabIndex={0}
        >
          <div
            onClick={[Function]}
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "justifyContent": "flex-start",
                "width": "initial",
              }
            }
          >
            <div
              className="makeStyles-colorCircle-2"
              style={
                Object {
                  "backgroundColor": "rgb(219, 64, 53)",
                }
              }
            />
            <span
              className="makeStyles-colorName-3"
            >
              Red
            </span>
          </div>
        </div>
        <input
          aria-hidden={true}
          autoFocus={false}
          className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
          disabled={false}
          onAnimationStart={[Function]}
          onChange={[Function]}
          required={false}
          tabIndex={-1}
          value="rgb(219, 64, 53)"
        />
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-186xcr5"
          >
            <span>
              Color
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
</div>
`;
